src/profile-logic/profile-data.js

```diff
@@ -715,6 +723,48 @@ function mapCallNodeSelectedStatesToSamples(
   return samplesSelectedStates;
 }
 
+/**
+ * The implementation of getSamplesSelectedStates for the inverted tree.
+ *
+ * This uses the suffix order, see the documentation of CallNodeInfoInverted.
+ */
+function _getSamplesSelectedStatesInverted(
+  sampleNonInvertedCallNodes: Array<IndexIntoCallNodeTable | null>,
+  activeTabFilteredNonInvertedCallNodes: Array<IndexIntoCallNodeTable | null>,
+  selectedInvertedCallNodeIndex: IndexIntoCallNodeTable,
+  callNodeInfo: CallNodeInfoInverted
+): SelectedState[] {
+  const suffixOrderIndexes = callNodeInfo.getSuffixOrderIndexes();
+  const [selectedSubtreeRangeStart, selectedSubtreeRangeEnd] =
+    callNodeInfo.getSuffixOrderIndexRangeForCallNode(
+      selectedInvertedCallNodeIndex
+    );
+  const sampleCount = sampleNonInvertedCallNodes.length;
+  const samplesSelectedStates = new Array(sampleCount);
+  for (let sampleIndex = 0; sampleIndex < sampleCount; sampleIndex++) {
+    let sampleSelectedState: SelectedState = 'SELECTED';
+    const callNodeIndex = sampleNonInvertedCallNodes[sampleIndex];
+    if (callNodeIndex !== null) {
+      const suffixOrderIndex = suffixOrderIndexes[callNodeIndex];
+      if (suffixOrderIndex < selectedSubtreeRangeStart) {
+        sampleSelectedState = 'UNSELECTED_ORDERED_BEFORE_SELECTED';
+      } else if (suffixOrderIndex >= selectedSubtreeRangeEnd) {
+        sampleSelectedState = 'UNSELECTED_ORDERED_AFTER_SELECTED';
+      }
+    } else {
```

Comment:
Looks like we also have no test covering inverted trees with samples filtered out...

(out of scope for the PR)